internal/executor/contracts/service_registry.go

```diff
@@ -0,0 +1,594 @@
+	}
+	if !sr.checkNameAvailable(name) {
+		return boltvm.Error(boltvm.BnsErrCode, fmt.Sprintf(string(boltvm.BnsErrMsg), fmt.Sprintf("The domain must register first")))
+	}
+	account := sr.GetAccount(sr.Caller()).(ledger.IAccount)
+	balance := account.GetBalance()
+
+	level1Domain := sr.getLevel1Domain()
+	if level1Domain[name]+GRACEPERIOD < uint64(sr.GetTxTimeStamp()/1000000000) || uint64(sr.GetTxTimeStamp()/1000000000)+duration+GRACEPERIOD < uint64(sr.GetTxTimeStamp()/1000000000)+GRACEPERIOD {
+		return boltvm.Error(boltvm.BnsErrCode, fmt.Sprintf(string(boltvm.BnsErrMsg), fmt.Sprintf("The domain id must registered in GRACEPERIOD ")))
+	}
+	price, err := sr.getPrice(name, level1Domain[name], duration)
+	if err != nil {
+		return boltvm.Error(boltvm.BnsErrCode, fmt.Sprintf(string(boltvm.BnsErrMsg), fmt.Sprintf("get register price: %v", err)))
+	}
+	registerCost := big.NewInt((int64)(price.premium + price.base))
+	if balance.Cmp(registerCost) < 0 {
+		return boltvm.Error(boltvm.BnsErrCode, fmt.Sprintf(string(boltvm.BnsErrMsg), fmt.Sprintf("Not enough Bitxhub Token provided")))
+	}
+	level1Domain[name] = level1Domain[name] + duration
+	sr.SetObject(Level1Domain, level1Domain)
+	account.SubBalance(registerCost)
+
+	return boltvm.Success(nil)
+}
+
+func (sr ServiceRegistry) AllocateSubDomain(parentName string, sonName string, owner string, resolver string) *boltvm.Response {
+	if parentName == "" {
+		return boltvm.Error(boltvm.BnsErrCode, fmt.Sprintf(string(boltvm.BnsErrMsg), "The domain name can not be an empty string"))
+	}
+	if !sr.checkNameAvailable(parentName) {
+		return boltvm.Error(boltvm.BnsErrCode, fmt.Sprintf(string(boltvm.BnsErrMsg), fmt.Sprintf("The domain must register first")))
+	}
+
+	if !checkBxhAddress(owner) {
+		return boltvm.Error(boltvm.BnsErrCode, fmt.Sprintf(string(boltvm.BnsErrMsg), fmt.Sprintf("The address is not valid")))
+	}
+
+	if resolver == "" || !sr.checkResolverAddress(resolver) {
+		return boltvm.Error(boltvm.BnsErrCode, fmt.Sprintf(string(boltvm.BnsErrMsg), fmt.Sprintf("The resolver is not in the list")))
+	}
+
+	if !sr.authorised(parentName) {
+		return boltvm.Error(boltvm.BnsErrCode, fmt.Sprintf(string(boltvm.BnsErrMsg), "The domain name does not belong to you"))
+	}
+
+	preRegister := make(map[string]bool)
+	sr.GetObject(PreRegister, &preRegister)
+	if preRegister[parentName] {
+		return boltvm.Error(boltvm.BnsErrCode, fmt.Sprintf(string(boltvm.BnsErrMsg), "current proposal is not end"))
+	}
+	subDomainProposalData := SubDomainProposalData{
+		ParentName:  parentName,
+		ParentOwner: sr.Caller(),
+		SonName:     sonName,
+		Onwer:       owner,
+		Resolver:    resolver,
+	}
+	subDomainProposalDataBytes, err := json.Marshal(subDomainProposalData)
+	if err != nil {
+		return boltvm.Error(boltvm.BnsErrCode, fmt.Sprintf(string(boltvm.BnsErrMsg), "marshal err"))
+	}
+	event := governance.EventRegister
+	proposalRes := sr.CrossInvoke(constant.GovernanceContractAddr.Address().String(), "SubmitProposal",
```

Comment:
why proposal needn't invoke function of 
sr.CrossInvoke(constant.GovernanceContractAddr.Address().String(), "ZeroPermission", pb.String(string(proposalRes.Result))) ?